src/services/webhook_service.py

```diff
@@ -0,0 +1,69 @@
+
+import json
+import requests
+import hashlib
+# Create a class with functions for getting a webhook url, token and id from a json file containing an array with webhook objects
+
+class WebhookService:
+
+    def __init__(self, webhook_file):
+        self.webhook_file = webhook_file
+
+    def get_webhook_by_id(self, webhook_id):
+        webhooks = self.get_webhooks()
+        for webhook in webhooks:
+            if webhook['id'] == webhook_id:
+                return webhook
+        return None
+
+    def create_payload_hash(self, payload, token):
+        x_payload_signature = hashlib.sha256(token.encode('utf-8'))
+        x_payload_signature.update(json.dumps(payload).encode('utf-8'))
+        return x_payload_signature.hexdigest()
+    
+    def create_webhook_payload(self, job_id, filename, url, success):
+        payload = {
+            "source": "waas",
+            "job_id": job_id,
+            "success": success
+        }
+        if success:
+            payload["url"] = url
+            payload["filename"] = filename
+        return payload
+    
+    def post_to_webhook(self, webhook_id, job_id, filename, url, success=False):
+        webhook = self.get_webhook_by_id(webhook_id) # This will return none if the webhook id is not found/not valid
+        if webhook:
+
+            payload = self.create_webhook_payload(job_id, filename, url, success)
+            payload_hash = self.create_payload_hash(payload, webhook['token'])
+            webhook_url = webhook['url']
+           
+            try:
+                req = requests.post(webhook_url, json=payload, allow_redirects=False, timeout=3, headers={'X-WAAS-Signature': payload_hash})
+                req.raise_for_status()
+            except requests.exceptions.Timeout as e:
+                print(f'Webhook request to {webhook_url} timed out')
+                raise
```

Comment:
Should this raise inside the catch? It would think it would just fail silent with the log?

Reply:
I see we do the same in email. I guess Sentry can catch it.

Reply:
It is just to provide a more informative error message. It will the re-raise the exception, so things will not fail silently. The "message and raise" pattern here https://www.loggly.com/blog/exceptional-logging-of-exceptions-in-python/

BTW: we should get rid of the print statements and get real logging. But that is for another PR.